castai/utils.go

```diff
@@ -17,6 +19,36 @@ func toStringPtr(value string) *string {
 	return &value
 }
 
+func toStringValue(value *string) string {
+	if value == nil {
+		return ""
+	}
+	return *value
+}
+
+func toPtr[S any](src S) *S {
+	return &src
+}
+
+func toStringMap(m map[string]interface{}) map[string]string {
+	out := make(map[string]string, len(m))
+	for k, v := range m {
+		out[k] = v.(string)
+	}
+	return out
+}
+
+func toStringList(items []interface{}) []string {
+	out := make([]string, 0, len(items))
+	for _, v := range items {
+		val, ok := v.(string)
+		if ok && val != "" {
```

Comment:
maybe check whitespace?

Reply:
there is validation on the field itself `validation.ToDiagFunc(validation.StringIsNotWhiteSpace)`